components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -171,6 +171,9 @@ public void addResidentIdP(IdentityProvider identityProvider, String tenantDomai
         if (saml2SSOResidentAuthenticatorConfig == null) {
             saml2SSOResidentAuthenticatorConfig = new FederatedAuthenticatorConfig();
             saml2SSOResidentAuthenticatorConfig.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.NAME);
+            saml2SSOResidentAuthenticatorConfig.setDefinedByType(IdentityConstants.DefinedByType.SYSTEM);
+            saml2SSOResidentAuthenticatorConfig.setAuthenticationType(
+                    IdentityConstants.AuthenticationType.IDENTIFICATION);
```

Comment:
Shouldn't these test cases should be working without setting these properties explicitly ?